src/main/java/com/exasol/containers/ExasolContainer.java

```diff
@@ -704,4 +704,14 @@ public final synchronized ExasolDriverManager getDriverManager() {
         }
         return this.driverManager;
     }
+
+    /**
+     * Get the IP address of the host running this container.
+     *
+     * @return IP address of the host
+     */
+    public String getHostIp() {
+        final HostIpDetector detector = new HostIpDetector(this);
```

Comment:
What if we create an instance of HostIpDetector once in the constructor and then use `getHostIp(this)` method?

Reply:
Also possible. I thought about whether that is necessary, but the method is seldom used and the detector has only one instance variable that needs to be initialized with a reference, so it is really cheap. In this case encapsulation wins over reuse-optimization.